sdk/core/System.Net.ClientModel/api/System.Net.ClientModel.net6.0.cs

```diff
@@ -0,0 +1,407 @@
+    }
+}
+namespace System.Net.ClientModel.Core
+{
+    public partial class ClientMessage : System.IDisposable
+    {
+        protected internal ClientMessage(System.Net.ClientModel.Core.MessageRequest request) { }
+        public virtual System.Threading.CancellationToken CancellationToken { get { throw null; } set { } }
+        public bool HasResponse { get { throw null; } }
+        public virtual System.Net.ClientModel.Core.MessageClassifier MessageClassifier { get { throw null; } set { } }
+        public virtual System.Net.ClientModel.Core.MessageRequest Request { get { throw null; } }
+        public virtual System.Net.ClientModel.Core.MessageResponse Response { get { throw null; } protected internal set { } }
+        public virtual void Dispose() { }
+        protected virtual void Dispose(bool disposing) { }
+        public void SetProperty(System.Type type, object value) { }
+        public bool TryGetProperty(System.Type type, out object? value) { throw null; }
+    }
+    [System.FlagsAttribute]
+    public enum ErrorBehavior
+    {
+        Default = 0,
+        NoThrow = 1,
+    }
+    public partial interface IJsonModel<out T> : System.Net.ClientModel.Core.IModel<T>
+    {
+        T Read(ref System.Text.Json.Utf8JsonReader reader, System.Net.ClientModel.ModelReaderWriterOptions options);
+        void Write(System.Text.Json.Utf8JsonWriter writer, System.Net.ClientModel.ModelReaderWriterOptions options);
+    }
+    public partial interface IModel<out T>
+    {
+        System.Net.ClientModel.ModelReaderWriterFormat GetWireFormat(System.Net.ClientModel.ModelReaderWriterOptions options);
+        T Read(System.BinaryData data, System.Net.ClientModel.ModelReaderWriterOptions options);
+        System.BinaryData Write(System.Net.ClientModel.ModelReaderWriterOptions options);
+    }
+    public partial class KeyCredentialAuthenticationPolicy : System.Net.ClientModel.Core.PipelinePolicy
+    {
+        public KeyCredentialAuthenticationPolicy(System.Net.ClientModel.KeyCredential credential, string headerName, string? keyPrefix = null) { }
+        public override void Process(System.Net.ClientModel.Core.ClientMessage message, System.Net.ClientModel.Core.PipelineEnumerator pipeline) { }
+        public override System.Threading.Tasks.ValueTask ProcessAsync(System.Net.ClientModel.Core.ClientMessage message, System.Net.ClientModel.Core.PipelineEnumerator pipeline) { throw null; }
+    }
+    public partial class MessageClassifier
+    {
+        protected internal MessageClassifier() { }
+        public virtual bool IsError(System.Net.ClientModel.Core.ClientMessage message) { throw null; }
+    }
+    public abstract partial class MessageHeaders
+    {
+        protected MessageHeaders() { }
+        public abstract int Count { get; }
+        public abstract void Add(string name, string value);
+        public abstract bool Remove(string name);
+        public abstract void Set(string name, string value);
```

Comment:
What would the behavior be here if the header doesn't exist? Should we consider making this TrySet for scenarios where the header doesn't exist?

Reply:
It should work like ```Dictionary<K,V>.this[K] = ...```, which does not throw.

Reply:
If it works, then `Add` seems redundant.

Reply:
Add throws if the key exists.

Reply:
I think the try patterns would be better here than relying on Exceptions.